### PR TITLE
Just like how caterpillars will turn into butterflies...

### PR DESCRIPTION
**Pull request by demipixel**
Players will turn into Entities.

### CODE REVIEW

**Comment by 109C** [comment]
In one ~~sweep~~ pull request the player is not longer unique in it's own class, being to reduced to a mere entity... Beautiful.

File: src/lib/plugins/blocks.js [hunk 1]
```diff
@@ -1,5 +1,3 @@
-var Vec3 = require("vec3").Vec3;
```
Comment (rom1504) [inline src/lib/plugins/blocks.js]: That line is needed (see the new Vec3 down there)

File: src/lib/plugins/entities.js [hunk 2]
```diff
@@ -76,6 +73,16 @@ module.exports.server=function(serv,options) {
         if (entity.deathTime && Date.now() - entity.bornTime >= entity.deathTime) {
           entity.destroy();
           return;
+        } else if (entity.pickupTime && Date.now() - entity.bornTime >= entity.pickupTime) {
+          var players = serv.getNearby({
+            world: entity.world,
+            position: entity.position,
+            radius: 1.5*32 // Seems good for now
+          });
+          if (players.length) {
```
Comment (rom1504) [inline src/lib/plugins/entities.js]: Why is that check needed ? UpdateAndSpawn already handle counting players and all.

File: src/lib/plugins/pvp.js [hunk 3]
```diff
@@ -15,17 +15,17 @@ module.exports.player=function(player,serv)
     if (!serv.entities[entityId]) return; // ?????
     var attackedPlayer = serv.entities[entityId].player;
```
Comment (rom1504) [inline src/lib/plugins/pvp.js]: entity.player doesn't exist anymore

File: src/lib/plugins/entities.js (hunk 2 above)
Comment (rom1504) [inline src/lib/plugins/entities.js]: Ah nevermind, I didn't understand what you were doing. You implemented collect, fun.

File: src/lib/behavior.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+module.exports = (obj) => {
+  return (eventName, data, func, opt) => {
+    var hiddenCancelled = false;
+    var cancelled = false;
+    var cancel = (hidden) => { // Hidden shouldn't be used often but it's not hard to implement so meh
```
Comment (rom1504) [inline src/lib/behavior.js]: what is hidden cancelled ?

File: src/lib/plugins/blocks.js (hunk 1 above)
Comment (demipixel) [inline src/lib/plugins/blocks.js]: Oops

File: src/lib/plugins/pvp.js (hunk 3 above)
Comment (demipixel) [inline src/lib/plugins/pvp.js]: Damnit. I still actually need to test the knockback stuff, haha.

File: src/lib/behavior.js (hunk 4 above)
Comment (demipixel) [inline src/lib/behavior.js]: It cancels the behavior without setting "cancelled" to true. This means you can trick other plugins into doing things. Really easy to implement for now, so if we don't find any use cases we can remove it.

Comment (rom1504) [inline src/lib/behavior.js]: ok

**Pull request merged**